corehq/apps/consumer_user/signals.py

```diff
@@ -1,75 +1,33 @@
-        try:
-            invitation = ConsumerUserInvitation.objects.filter(case_id=case.case_id,
-                                                               domain=case.domain,
-                                                               active=True).latest('invited_on')
-            if invitation:
-                if updated_by and updated_by == 'patient' and email == invitation.email and not case.closed:
-                    return
-                invitation.active = False
-                ConsumerUserCaseRelationship.objects.filter(case_id=case.case_id,
-                                                            domain=case.domain).delete()
-                invitation.save()
-        except ConsumerUserInvitation.DoesNotExist:
-            pass
-        if case.closed:
-            return
-        invitation = ConsumerUserInvitation.objects.create(case_id=case.case_id, domain=case.domain)
-        invitation.invited_by = case.opened_by
-        invitation.email = email
-        invitation.accepted = False
-        invitation.save()
-        # Change to https after the discussion
-        url = 'http://%s%s' % (get_site_domain(),
-                               reverse('consumer_user:patient_register',
-                                       kwargs={
-                                           'invitation': TimestampSigner().sign(urlsafe_base64_encode(
-                                               force_bytes(invitation.pk)
-                                           ))
-                                       }))
-        email_context = {
-            'link': url,
-        }
-        send_html_email_async.delay(
-            'Beneficiary Registration',
-            email,
-            render_to_string('email/registration_email.html', email_context),
-            text_content=render_to_string('email/registration_email.txt', email_context)
-        )
-        update_case(case.domain, case.case_id, {'invitation_status': 'sent', 'updated_by': 'patient'})
+def make_invitation_inactive(invitation, case):
+    invitation.make_inactive()
+    ConsumerUserCaseRelationship.objects.filter(case_id=case.case_id,
+                                                domain=case.domain).delete()
+
 
-    except Exception:
-        notify_exception(
-            None,
-            message="Could not create messaging case changed task. Is RabbitMQ running?"
-        )
+def send_email_case_changed_receiver(sender, case, **kwargs):
+    if case.type != CONSUMER_INVITATION_CASE_TYPE:
+        return
+    email = case.get_case_property('email')
+    invitation = ConsumerUserInvitation.objects.filter(case_id=case.case_id,
+                                                       domain=case.domain,
+                                                       active=True).last()
+    if case.closed:
+        if invitation:
+            make_invitation_inactive(invitation, case)
+        return
+    elif invitation and email == invitation.email:
```

Comment:
Is there a way we can resend an invitation in this case?

Reply:
Made new changes to signals logic. If we set the status in case_properties to anything other than sent or accepted it should create a new invitation and send a new mail.